### ISSUE TITLE
Limit and reduce use of OpenSim::Array

### ISSUE THREAD

**Issue by aseth1** [opensim-org/opensim-core#1194]
In addition to OpenSim::Set there is wide use of OpenSim:;Array. We should eventually move to exclusive use of `std::` containers so that we don't have to maintain unnecessary code and run into unexpected bugs (e.g. #1119) from less than ideal implementations.

Also came up in #1146 where our test utilities use OpenSim::Array for containing tolerances:
>  OpenSim::Arrays are used to define the tolerances as expected by CHECK_STORAGE_AGAINST_STANDARD() and updating that method would impact numerous test cases. Making that update perhaps should wait until we are closer to eliminating OpenSim::Array altogether

**Comment by carmichaelong** [comment]
@aseth1 I looked at `CHECK_STORAGE_AGAINST_STANDARD()` and tests using this, and it doesn't look too time consuming to change it to use `std::vector` instead. If you think that we should just do that (instead of backing `OpenSim::Array` with `std::vector`) I can work on it.

**Comment by chrisdembia** [comment]
@klshrinidhi has worked on replacing `OpenSim::Array` with `std::vector`. He may have input here.

**Comment by klshrinidhi** [comment]
I have a local branch on this. It is a pretty time consuming task to replace OpenSim::Array with std::vector completely. I stopped working on it several weeks ago and my branch now is severely out of sync with master. You can go ahead if you have the time. I am probably not going to do this anytime soon.

**Comment by carmichaelong** [comment]
> It is a pretty time consuming task to replace OpenSim::Array with std::vector completely.

I agree that to **completely** do it would take quite awhile, and shouldn't be done all in one PR.

It seemed to me that for this case specifically, changing `CHECK_STORAGE_AGAINST_STANDARD` is a low hanging fruit that would also be efficient in preventing further use of OpenSim::Array. I think I'll go ahead and work on this specific case then.

**Comment by jenhicks** [comment]
Back OpenSim Array with std::vec
